python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -408,10 +411,45 @@ def _add_task(self, task_cb, priority, task_args=None, task_kwargs=None):
                                               task_kwargs = task_kwargs)
         return str(task_id)
 
+    def request_attachment(self, attachment_entity):
+        """
+        Downloads an attachment from Shotgun asynchronously or returns a cached
+        file path if found.
+
+        :param dict attachment_entity: The Attachment entity to download data from.
```

Comment:
what dict fields are required here? Looked at the code and looks like it needs a `this_file` key - would be great with some details and an example. Do we need to be concerned with S3 ids timing out?